test/controller.test.ts

```diff
@@ -1502,6 +1502,398 @@ contract('Controller', ([owner, accountOwner1, accountOperator1, random]) => {
     })
   })
 
+  describe('Short otoken', () => {
+    let longOtoken: MockOtokenInstance
+    let shortOtoken: MockOtokenInstance
+
+    before(async () => {
+      const expiryTime = new BigNumber(60 * 60 * 24) // after 1 day
+
+      longOtoken = await MockOtoken.new()
+      shortOtoken = await MockOtoken.new()
+      // init otoken
+      await longOtoken.init(
+        addressBook.address,
+        weth.address,
+        usdc.address,
+        usdc.address,
+        new BigNumber(250).times(new BigNumber(10).exponentiatedBy(18)),
+        new BigNumber(await time.latest()).plus(expiryTime),
+        true,
+      )
+      await shortOtoken.init(
+        addressBook.address,
+        weth.address,
+        usdc.address,
+        usdc.address,
+        new BigNumber(200).times(new BigNumber(10).exponentiatedBy(18)),
+        new BigNumber(await time.latest()).plus(expiryTime),
+        true,
+      )
+
+      // give free money
+      await longOtoken.mintOtoken(accountOwner1, new BigNumber('100'))
+      await longOtoken.mintOtoken(accountOperator1, new BigNumber('100'))
+      await usdc.mint(accountOwner1, new BigNumber('1000000'))
+      await usdc.mint(accountOperator1, new BigNumber('1000000'))
+      await usdc.mint(random, new BigNumber('1000000'))
+    })
+
+    describe('Mint short otoken', () => {
+      it('should revert minting from random address other than owner or operator', async () => {
+        const vaultCounter = new BigNumber(await controller.getAccountVaultCounter(accountOwner1))
+        assert.isAbove(vaultCounter.toNumber(), 0, 'Account owner have no vault')
+
+        const collateralToDeposit = new BigNumber(await shortOtoken.strikePrice()).dividedBy(1e18)
+        const amountToMint = new BigNumber('1')
+        const actionArgs = [
+          {
+            actionType: ActionType.MintShortOption,
+            owner: accountOwner1,
+            sender: random,
+            asset: shortOtoken.address,
+            vaultId: vaultCounter.toNumber(),
+            amount: amountToMint.toNumber(),
+            index: '0',
+            data: ZERO_ADDR,
+          },
+          {
+            actionType: ActionType.DepositCollateral,
+            owner: accountOwner1,
+            sender: random,
+            asset: usdc.address,
+            vaultId: vaultCounter.toNumber(),
+            amount: collateralToDeposit.toNumber(),
+            index: '0',
+            data: ZERO_ADDR,
+          },
```

Comment:
We can remove this action? (It looks like the test should still fail)

Reply:
Yeah I don't know what that action is doing there 👍